### PR TITLE
ea 17 checkout page epic

### LINKED ISSUE #17: Checkout page epic
**Is your feature request related to a problem? Please describe.**
As a user, the checkout page should display the sale calculations, user input for the address and payment information, and buttons to return to the cart page and place the order.

Calculations include:
- subtotal
- tax based on state
- shipping and handling
- grand total

**Describe the solution you'd like**
- Show inputs for address and payment
- Show button to return to the cart page
- Show button to place the order
- Make page scrollable to see everything

Epic Issues:
- [x] #18 
- [x] #22 
- [x] #20 
- [x] #24

### PR DESCRIPTION
**Pull request by eangele1**
## Changes
1. Added tax + price calculations to checkout screen
2. Added colors file to be utilized throughout the project
3. Updated cart item ID type to "number"

## Purpose
The checkout page will bring in the data from the cart screen and take in inputs for shipping/billing and payment and display the total costs from the data provided.

## Approach
Checkout page now accepts user input for addresses and payment and utilizes Redux state to calculate the grand total which includes shipping & handling (5.99 by default) and sales tax from zip code.

## Testing Steps
By adding items into the cart from previous screens, the checkout screen can use the information to display the total costs.

Closes #17

### CODE REVIEW

File: data/colors.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const colors = {
+  tan: "#dcb688",
+  violet: "#814e66",
+  black: "#000",
+  white: "#FFF",
+};
+
+export default colors;
```
Comment (BeakalDegefa) [inline data/colors.ts:8]: I like this, Thank you Ezer.

File: data/states.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    [5001, 5495],
+    [5601, 5907],
+  ],
+  WA: [98001, 99403],
+  WI: [53001, 54990],
+  WV: [24701, 26886],
+  WY: [82001, 83128],
+};
+
+export default states;
```
Comment (BeakalDegefa) [inline data/states.ts]: This is good.

File: data/colors.ts (hunk 1 above)
Comment (yelab07) [inline data/colors.ts:8]: Thanks for adding the color component ,this will help a lot !

File: data/taxRates.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  UT: 0.061,
+  VA: 0.53,
+  VT: 0.06,
+  WA: 0.065,
+  WI: 0.05,
+  WV: 0.06,
+  WY: 0.04,
+};
+
+export default taxRates;
```
Comment (BeakalDegefa) [inline data/taxRates.ts]: Good work. I would name the variable StateTaxRate.

File: data/states.ts (hunk 2 above)
Comment (yelab07) [inline data/states.ts]: Are this zip codes for shipment purpose ?I think it would be better to specify so it is clear for everyone .

**Review by BeakalDegefa — APPROVED** [review]

File: redux/cartSlice.ts [hunk 4]
```diff
@@ -1,42 +1,46 @@
...
 export const cartReducer = cartSlice.reducer;
-export const {
-  addToCart,
-  incrementQuantity,
-  decrementQuantity,
-  removeItem,
-} = cartSlice.actions;
\ No newline at end of file
+export const { addToCart, incrementQuantity, decrementQuantity, removeItem } =
+  cartSlice.actions;
```
Comment (yelab07) [inline redux/cartSlice.ts:46]: Great Job Ezer creating those slices !

File: screens/Cart/CheckoutScreen.tsx [hunk 5]
```diff
@@ -1,16 +1,276 @@
...
+  },
+  longInput: {
+    backgroundColor: colors.white,
+    paddingHorizontal: 10,
+    width: 375,
+    marginVertical: 5,
+    borderRadius: 10,
+    elevation: 10,
+  },
 });
```
Comment (yelab07) [inline screens/Cart/CheckoutScreen.tsx:154]: I like the logic you implemented in here .I  think using and  creating one or more component will make rendering it here would make the code in this page shorter and neat as you can also  avoid the long inline styles lines you have in here .

**Review by yelab07 — CHANGES_REQUESTED** [review]
Great work Ezer please check on the feedback's I left ,thanks a lot for the amazing work you are doing !

File: data/taxRates.ts (hunk 3 above)
Comment (eangele1) [reply data/taxRates.ts]: Noted!

File: data/states.ts (hunk 2 above)
Comment (eangele1) [reply data/states.ts]: Yes, I've laid out a comment within the file that explains what the zip codes are for.

File: screens/Cart/CheckoutScreen.tsx (hunk 5 above)
Comment (eangele1) [reply screens/Cart/CheckoutScreen.tsx:154]: Noted!

**Review by yelab07 — APPROVED** [review]
Great job Ezer I really liked the changes you made by adding new components !!

**Pull request merged**